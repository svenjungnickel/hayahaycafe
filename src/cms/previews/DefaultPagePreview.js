import React from 'react';
import PropTypes from 'prop-types';
import DefaultPageTemplate from '../../templates/DefaultPageTemplate';

const DefaultPagePreview = ({ entry, widgetFor }) => (
    <DefaultPageTemplate
        title={entry.toJS().data.title}
        subtitle={entry.toJS().data.subtitle}
        headerImage={entry.toJS().data.headerImage}
        content={widgetFor('body')}
    />
);

DefaultPagePreview.propTypes = {
    entry: PropTypes.shape({
        getIn: PropTypes.func,
    }),
    widgetFor: PropTypes.func,
};

export default DefaultPagePreview;
