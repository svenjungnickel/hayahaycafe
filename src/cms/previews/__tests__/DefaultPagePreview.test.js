import React from 'react';
import renderer from 'react-test-renderer';
import DefaultPagePreview from '../DefaultPagePreview';

jest.mock('../../../components/Header');

describe('DefaultPagePreview', () => {
    it('renders without entry throws prop type error', () => {
        const renderDefaultPagePreview = () => {
            const widgetFor = () => {};

            const component = <DefaultPagePreview widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderDefaultPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders without widgetFor throws prop type error', () => {
        const renderDefaultPagePreview = () => {
            const entry = {
                toJS: () => {},
            };

            const component = <DefaultPagePreview entry={entry} />;
            renderer.create(component);
        };

        expect(renderDefaultPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid entry throws prop type error', () => {
        const renderDefaultPagePreview = () => {
            const entry = {
                toJS: 123,
            };
            const widgetFor = () => {};

            const component = <DefaultPagePreview entry={entry} widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderDefaultPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid widgetFor throws prop type error', () => {
        const renderDefaultPagePreview = () => {
            const entry = {
                toJS: () => {},
            };
            const widgetFor = {};

            const component = <DefaultPagePreview entry={entry} widgetFor={widgetFor} />;
            renderer.create(component);
        };

        expect(renderDefaultPagePreview).toThrowError('Warning: Failed prop type');
    });

    it('renders correctly', () => {
        const entry = {
            toJS: () => {
                return {
                    data: {
                        title: 'title',
                        subtitle: 'subtitle',
                        headerImage: {
                            src: 'src',
                        },
                    },
                };
            },
        };
        const widgetFor = () => 'content';

        const component = <DefaultPagePreview entry={entry} widgetFor={widgetFor} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
