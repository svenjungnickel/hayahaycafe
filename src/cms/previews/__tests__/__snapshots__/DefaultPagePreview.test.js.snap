// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DefaultPagePreview renders correctly 1`] = `
[
  <header />,
  <section
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div>
            content
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;
